core_lib/data_transform/result_to_dict.py

```diff
@@ -100,6 +106,10 @@ def result_to_dict(return_val, properties_as_dict: bool = True, callback: Callab
     else:
         results = return_val
 
+    if isinstance(return_val, mongomock.collection.Cursor):

```

Comment:
here you are checking an instance that comes from `mongomock` , and we use it just for testing, to check if it will work with the data that the real Mongodb connection returns us and make sure to add it over here